DQM/SiTrackerPhase2/plugins/Phase2ITMonitorRecHit.cc

```diff
@@ -0,0 +1,390 @@
+  edm::LogInfo("Phase2ITMonitorRecHit") << ">>> Construct Phase2ITMonitorRecHit ";
+}
+
+Phase2ITMonitorRecHit::~Phase2ITMonitorRecHit() {
+  edm::LogInfo("Phase2ITMonitorRecHit") << ">>> Destroy Phase2ITMonitorRecHit ";
+}
+// -- Analyze
+void Phase2ITMonitorRecHit::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) { fillITHistos(iEvent); }
+
+void Phase2ITMonitorRecHit::fillITHistos(const edm::Event& iEvent) {
+  // Get the RecHits
+  edm::Handle<SiPixelRecHitCollection> rechits;
+  iEvent.getByToken(tokenRecHitsIT_, rechits);
+  if (!rechits.isValid())
+    return;
+  std::map<std::string, unsigned int> nrechitLayerMap;
+  unsigned long int nTotrechitsinevt = 0;
+  // Loop over modules
+  SiPixelRecHitCollection::const_iterator DSViter;
+  for (DSViter = rechits->begin(); DSViter != rechits->end(); ++DSViter) {
+    // Get the detector id
+    unsigned int rawid(DSViter->detId());
+    DetId detId(rawid);
+    // Get the geomdet
+    const GeomDetUnit* geomDetunit(tkGeom_->idToDetUnit(detId));
+    if (!geomDetunit)
+      continue;
+    std::string key = Phase2TkUtil::getITHistoId(detId.rawId(), tTopo_);
+    nTotrechitsinevt += DSViter->size();
+    if (nrechitLayerMap.find(key) == nrechitLayerMap.end()) {
+      nrechitLayerMap.insert(std::make_pair(key, DSViter->size()));
+    } else {
+      nrechitLayerMap[key] += DSViter->size();
+    }
+
+    edmNew::DetSet<SiPixelRecHit>::const_iterator rechitIt;
+    //loop over rechits for a single detId
+    for (rechitIt = DSViter->begin(); rechitIt != DSViter->end(); ++rechitIt) {
+      LocalPoint lp = rechitIt->localPosition();
+      Global3DPoint globalPos = geomDetunit->surface().toGlobal(lp);
+      //in mm
+      double gx = globalPos.x() * 10.;
+      double gy = globalPos.y() * 10.;
+      double gz = globalPos.z() * 10.;
+      double gr = globalPos.perp() * 10.;
+      //Fill global positions
+      if (geomDetunit->subDetector() == GeomDetEnumerators::SubDetector::P2PXB) {
+        globalXY_barrel_->Fill(gx, gy);
+        globalRZ_barrel_->Fill(gz, gr);
+      } else if (geomDetunit->subDetector() == GeomDetEnumerators::SubDetector::P2PXEC) {
+        globalXY_endcap_->Fill(gx, gy);
+        globalRZ_endcap_->Fill(gz, gr);
+      }
+      //layer wise histo
+      if (layerMEs_[key].clusterSizeX)
+        layerMEs_[key].clusterSizeX->Fill(rechitIt->cluster()->sizeX());
+      if (layerMEs_[key].clusterSizeY)
+        layerMEs_[key].clusterSizeY->Fill(rechitIt->cluster()->sizeY());
+      if (layerMEs_[key].globalPosXY)
+        layerMEs_[key].globalPosXY->Fill(gx, gy);
+      if (layerMEs_[key].globalPosRZ)
+        layerMEs_[key].globalPosRZ->Fill(gz, gr);
+      if (layerMEs_[key].localPosXY)
+        layerMEs_[key].localPosXY->Fill(lp.x(), lp.y());
+      if (layerMEs_[key].posX)
+        layerMEs_[key].posX->Fill(lp.x());
+      if (layerMEs_[key].posY)
+        layerMEs_[key].posX->Fill(lp.y());
```

Comment:
> RecHit_Y seems empty, despite recHit_Y_error is not

Indeed this is buggy.

```suggestion
        layerMEs_[key].posY->Fill(lp.y());
```

Reply:
corrected